src/app/public/modules/inline-form/inline-form.component.ts

```diff
@@ -0,0 +1,236 @@
+              }
+            ]);
+          });
+        break;
+
+      case SkyInlineFormType.SaveCancel:
+        Observable.zip(
+          this.resourcesService.getString('skyux_inline_form_button_save'),
+          this.resourcesService.getString('skyux_inline_form_button_cancel')
+        ).subscribe((values: any) => {
+          emitter.next([
+            {
+              text: values[0],
+              styleType: 'primary',
+              action: 'save'
+            },
+            {
+              text: values[1],
+              styleType: 'link',
+              action: 'cancel'
+            }
+          ]);
+        });
+        break;
+
+      case SkyInlineFormType.DoneDeleteCancel:
+        Observable.zip(
+          this.resourcesService.getString('skyux_inline_form_button_done'),
+          this.resourcesService.getString('skyux_inline_form_button_delete'),
+          this.resourcesService.getString('skyux_inline_form_button_cancel')
+
+        )
+          .subscribe((values: any) => {
+            emitter.next([
+              {
+                text: values[0],
+                styleType: 'primary',
+                action: 'done'
+              },
+              {
+                text: values[1],
+                styleType: 'default',
+                action: 'delete'
+              },
+              {
+                text: values[2],
+                styleType: 'link',
+                action: 'cancel'
+              }
+            ]);
+          });
+        break;
+
+      case SkyInlineFormType.SaveDeleteCancel:
+        Observable.zip(
+          this.resourcesService.getString('skyux_inline_form_button_save'),
+          this.resourcesService.getString('skyux_inline_form_button_delete'),
+          this.resourcesService.getString('skyux_inline_form_button_cancel')
+
+        )
+          .subscribe((values: any) => {
+            emitter.next([
+              {
+                text: values[0],
+                styleType: 'primary',
+                action: 'save'
+              },
+              {
+                text: values[1],
+                styleType: 'default',
+                action: 'delete'
+              },
+              {
+                text: values[2],
+                styleType: 'link',
+                action: 'cancel'
+              }
+            ]);
+          });
+        break;
+    }
+
+    return emitter;
+  }
+
+  private getCustomButtons(buttonConfig: SkyInlineFormButtonConfig[]): SkyInlineFormButton[] {
+    const buttons: SkyInlineFormButton[] = [];
+
+    buttonConfig.forEach((config: SkyInlineFormButtonConfig) => {
+      buttons.push({
+        text: config.text,
+        action: config.action,
+        styleType: config.styleType || 'default'
+      } as SkyInlineFormButton);
+    });
+
+    return buttons;
+  }
+
+  private isValidCustomConfig(config: SkyInlineFormConfig): boolean {
+    return (
+      config
+      && config.type === SkyInlineFormType.Custom
```

Comment:
This is personal preference, but I've always liked the operator on the tail-end of the new line. It lines up the variables vertically and I find it easier to read. What do you think?

Either way, we should probably update the style guide with the team's decision.
```
return (
  config &&
  config.type === SkyInlineFormType.Custom &&
  config.buttons.length > 0
);
```